### PR TITLE
#1098 Add `configuratorType` state variable to allow CSR mode

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-white/issues/1098 |
| Dependencies | - https://github.com/ripe-tech/ripe-sdk/pull/488 |
| Decisions | - Add `configuratorType` state variable and adapt configurator style to allow `CSR` type |

### CODE REVIEW

File: vue/components/organisms/configurator/configurator.vue [hunk 1]
```diff
@@ -486,6 +360,177 @@ export const Configurator = {
...
+                if (this.configurator && this.configurator.ready) {
+                    this.configurator.lowlight(part);
+                }
+            });
+
+            this.resize(this.size, this.width, this.height);
+
+            if (this.configuratorType === "csr") {
+                this.ripeInstance.trigger("post_config", this.ripeInstance.loadedConfig);
+            }
```
Comment (NFSS10) [inline vue/components/organisms/configurator/configurator.vue:528]: Why is this needed? It's because of white not using ripe-sdk correctly right?

**Review by NFSS10 — CHANGES_REQUESTED** [review]

File: vue/components/organisms/configurator/configurator.vue [hunk 2]
```diff
@@ -486,6 +360,177 @@ export const Configurator = {
...
+            this.configurator.bind("highlighted_part", part => {
+                this.$bus.trigger("highlighted_part", this.configurator, part);
+            });
+
+            this.configurator.bind("lowlighted", () => {
+                this.$bus.trigger("lowlighted", this.configurator);
+            });
+
+            this.configurator.bind("ready", options => {
+                if (options?.origin && options.origin.includes("configurator")) {
```
[the remark this answers is not in the document]
Comment (3rdvision) [reply vue/components/organisms/configurator/configurator.vue:434]: > If you are using optional chaining you can do this options?.origin?.includes("configurator")
You can't do that when you're calling a method because it will throw

> Also, I think you should look for a more specific origin as in, use .startsWith() (it is also more efficient)
Will do

File: vue/components/organisms/configurator/configurator.vue (hunk 1 above)
Comment (3rdvision) [reply vue/components/organisms/configurator/configurator.vue:528]: This is because in ripe-white implementation `post_config` is triggered before the `_registerConfigHandlers` method in CSR which CSR uses to initialize the initials and render.

This has to do with the fact that we're binding a CSR configurator after the `post_config` already happened so we're triggering a `post_config` manually.

image

I improved the guard and added a comment to make it clear but feel free to share if you have any suggestion on improving this

File: vue/components/organisms/configurator/configurator.vue [hunk 3]
```diff
@@ -523,7 +523,14 @@ export const Configurator = {
...
-            if (this.configuratorType === "csr") {
+            // in case the ripe instance already loaded the config
+            // and the CSR configurator is not yet loaded
+            // re-trigger `post_config` event for the CSR configurator
+            if (
+                this.ripeInstance.loadedConfig &&
+                this.configuratorType === "csr" &&
+                this.configurator.loading
+            ) {
                 this.ripeInstance.trigger("post_config", this.ripeInstance.loadedConfig);
```
Comment (NFSS10) [inline vue/components/organisms/configurator/configurator.vue:534]: The `trigger` method is `async`, is it intended to not have an await here?

**Review by NFSS10 — CHANGES_REQUESTED** [review]

File: vue/components/organisms/configurator/configurator.vue (hunk 3 above)
Comment (3rdvision) [reply vue/components/organisms/configurator/configurator.vue:534]: Yes, this is intended
